tests/tensor/test_extra_ops.py

```diff
@@ -482,6 +474,22 @@ def test_scalar_input(self):
         ):
             squeeze(x, axis=1)
 
+    def test_invalid_input(self):
+        x = pt.vector("x")
+        axis = 0
+
+        f = pytensor.function([x], pt.squeeze(x, axis))
+
+        # Test that we allow squeezing of valid non-broadcastable dimension
+        assert f([0]) == 0
+
+        # Test that we cannot squeeze dimensions whose ength is greater than 1
+        with pytest.raises(
+            AssertionError,
```

Comment:
The is failing because it doesn't raise an AssertionError but a ValueError?

Reply:
It does raise an assertion error. The regex pattern does not match strangely when using python 3.12 as evident [here](https://github.com/pymc-devs/pytensor/actions/runs/8750088177/job/24012954012?pr=713#step:6:4121) .

Reply:
Is it version specific or do we just need to tweak the regex?

Reply:
It is version specific. Regex exp works fine with my local python v3.11.8

Reply:
You may not be running locally with the same compile flag. Add this at the top of the test file and see if you can reproduce locally then:

```python
import pytensor
pytensor.config.mode = "FAST_COMPILE"
```
Note that's the flag in the failing test: `Test py3.12 : fast-compile 1 `
This is a bit like the `floatX="float32"`, because it's not the default you may be surprised the test is failing remotely but not locally.

I think this happens because in fast compile we're running the python implementation of `SpecifyShape` and in the default mode we're running the C implementation which have slightly different error messages. Can you make the match statement handle both cases?